src/dbnode/storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -485,23 +329,28 @@ func (s *fileSystemSource) loadShardReadersDataIntoShardResult(
 	runOpts bootstrap.RunOptions,
 	runResult *runResult,
 	ropts result.Options,
-	timeWindowReaders timeWindowReaders,
-	readerPool *readerPool,
+	timeWindowReaders bootstrapper.TimeWindowReaders,
+	readerPool *bootstrapper.ReaderPool,
 ) {
 	var (
-		blockPool         = ropts.DatabaseBlockOptions().DatabaseBlockPool()
-		seriesCachePolicy = ropts.SeriesCachePolicy()
-		indexBlockSegment segment.MutableSegment
-		timesWithErrors   []time.Time
-		nsCtx             = namespace.NewContextFrom(ns)
+		blockPool                  = ropts.DatabaseBlockOptions().DatabaseBlockPool()
+		seriesCachePolicy          = ropts.SeriesCachePolicy()
+		indexBlockDocumentsBuilder segment.DocumentsBuilder
+		timesWithErrors            []time.Time
+		nsCtx                      = namespace.NewContextFrom(ns)
+		docsPool                   = s.opts.IndexOptions().DocumentArrayPool()
+		batch                      = docsPool.Get()
 	)
+	defer func() {
+		docsPool.Put(batch)
+	}()
```

Comment:
nit: Realize this can be abbreviated to `defer docsPool.Put(batch)` (think this code was elsewhere, but just FYI).